lib/index.js

```diff
@@ -0,0 +1,151 @@
+/**!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brPassport = require('bedrock-passport');
+const {validate} = require('bedrock-validation');
+const {asyncHandler} = require('bedrock-express');
+const paymentService = require('bedrock-payment');
+const logger = require('./logger');
+
+require('./config.js');
+
+const {PaymentStatus, Errors} = paymentService;
+
+const {ensureAuthenticated} = brPassport;
+const {config} = bedrock;
+const {BedrockError} = bedrock.util;
+
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  const cfg = config['payment-http'];
+  const {basePath: PAYMENT_ENDPOINT} = cfg.routes;
+  // GET /payment/credentials
+  // GET the api keys/ auth tokens necessary to implement charges client side.
+  app.get(
+    `${PAYMENT_ENDPOINT}/credentials`,
+    ensureAuthenticated,
+    asyncHandler(async (req, res) => {
+      const {service} = req.query;
+      if(!service) {
+        throw new BedrockError(
+          'You must provide a service in the query', Errors.Syntax);
+      }
+      const data = await paymentService.getGatewayCredentials({service});
```

Comment:
Seems like this API should throw a `NotFoundError` instead of returning undefined/null as shown here.

Reply:
That makes the code less re-usable by other services and it also hard codes an error when maybe we don't want to throw one. I actually really dislike this pattern of throwing on NotFound and it causes problems for me earlier because the balance service throws an error when a user doesn't have a balance yet. I guess my point is we can't guarantee that in every situation a payment gateway not having credentials is really an error. So I would prefer to keep the errors inside of the handler and separate them from function.

Reply:
There is a well established convention for `get` APIs to throw NotFoundError.  If you want to provide special handing for a NotFoundError you simply surround the API call with a try/catch and test `error.name === 'NotFoundError'`.

I don't care for APIs that required you to test for the very existence of a return value.  Also, this goes against using destructed assignment.  I like the contract with the API to be such that I can do the following without fear of a destructured assignment error when the API returns `undefined`.
```
const {foo} = await bar.getSomething();
```

Reply:
Can we agree to disagree here? I think programming inside of try/catch loops is horrible style and kind of defeats the purpose of try/catch. Catch, for me at least, is where you log an error and admit defeat, not where you add code to handle a situation you were expecting.

Reply:
throwing errors is the standard pattern for error handling with Promises and `try/catch` is the standard pattern for error handling with `async/await`.

Is it the case that @dlongley is providing the paymentService API, if so, we can see what he comes up with.

Reply:
A use case you might not have though of is a user requests multiple credentials. In this case `null` would mean that one of those payment gateway credentials would fail, but another payment credential might be valid. I guess I just feel like throwing inside of this function hems it in to strongly and to soon.

Reply:
I might add on other projects I was expecting `null` and got an error thrown. What the absence of something means is really defined by the parent content. In this case the handler says `null` means an error in another it could mean something else.

Reply:
Addressed in https://github.com/digitalbazaar/bedrock-payment-http/pull/1/commits/bd4395f747d33301e80694d02b24d0b7010cf706